From 5b742529f8e6083c83340e8cfdae373ec0ddea04 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 15 Jun 2018 10:41:47 +0800
Subject: [PATCH 4050/5242] MLK-18618-2: hdp: Recovery pre video mode after
 cable plugin

commit  94dd4efbf7d975f3b4392abcf00613e6333a94fe from
https://source.codeaurora.org/external/imx/linux-imx.git

The patch can fix 4Kp60 video mode failed to work afer cable
plugout/in.
DRM driver will not call mode set function if video mode isn't changed.
But for 4Kp60, scrambling and TMDS_Bit_Clock_Ratio should
reconfiguration after cable plugin.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index c768557..5f9797b 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -994,6 +994,10 @@ static void hotplug_work_func(struct work_struct *work)
 
 	if (connector->status == connector_status_connected) {
 		/* Cable Connected */
+		/* For HDMI2.0 SCDC should setup again.
+		 * So recovery pre video mode if it is 4Kp60 */
+		if (drm_mode_equal(&hdp->video.pre_mode, &edid_cea_modes[3]))
+			imx_hdp_mode_setup(hdp, &hdp->video.pre_mode);
 		DRM_INFO("HDMI/DP Cable Plug In\n");
 		enable_irq(hdp->irq[HPD_IRQ_OUT]);
 	} else if (connector->status == connector_status_disconnected) {
-- 
1.7.9.5

